### DISCUSSION TITLE
Swift code debugging

### DISCUSSION THREAD

**Discussion by alloc33** [mfussenegger/nvim-dap#570] [Q&A]
Is there any way to configure LLDB for swift debugging? I just moved from vscode but I can't find any examples how to configure neovim for swift debugging.

I got this while trying to run codelldb on swift file:
```
Debug adapter didn't respond. Either the adapter is slow (then wait and ignore this) or there is a problem with your adapter or `debugserver` configuration. Check the logs for errors (:help dap.set_log_level)
```

my config:
```lua
local dap = require('dap')

dap.configurations.swift = {
	{
		name = "Launch file",
		type = "debugserver",
		request = "launch",
		program = function()
			return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
		end,
		cwd = '${workspaceFolder}',
		stopOnEntry = true,
	},
}
dap.adapters.debugserver = {
	type = 'server',
	host = '127.0.0.1',
	port = 13000 -- ?? Use the port printed out or specified with `--port`
}
```

This is what happening in terminal:
```
debugserver --port 13000
debugserver: unrecognized option `--port'
debugserver-@(#)PROGRAM:debugserver  PROJECT:debugserver-1000.11.38.2
 for x86_64.
Listening to port 13000 for a connection from 127.0.0.1...
Got a connection, waiting for process information for launching or attaching.
```

And nothing happens.. How to make it work on swift files?

**Comment by Santhosh-KS** [comment]
@xbladesub  did you get dap working for swift in any ways? If so can you please post some findings here?

**Reply by wojciech-kulik** [reply]
Check out my plugin: https://github.com/wojciech-kulik/xcodebuild.nvim - it might be helpful

**Reply by Santhosh-KS** [reply]
Thank you for your response. Some how I missed your message. I had stopped on this work. But now I'm resuming back on it. I did look at your repo. Looks promising. I will try it out and keep you posted. On the first outlook I understand that in that repo you are trying to give all the features of xcode available on neovim! thats really awesome. I'm mainly focussing on the serverside swift aspects of it and working on linux. Not all of them I may need from the repo. I will pick the dap part of it. Before that I just include your repo and see what sort of errors I get. I will keep you posted in this thread.

**Reply by Santhosh-KS** [reply]
Hi @wojciech-kulik  I tried out the DAP things from your plugin on Linux. It works great! Thanks again for your help and amazing work on the xcodebuild.nvim project!

**Reply by wojciech-kulik** [reply]
Great to hear that it worked! :)

**Reply by vimscientist69** [reply]
> Hi @wojciech-kulik I tried out the DAP things from your plugin on Linux. It works great! Thanks again for your help and amazing work on the xcodebuild.nvim project!

@Santhosh-KS , can you please share what you did? I want to setup debugger for Swift server side as well.
